C#/Fold.tmPreferences

```diff
@@ -0,0 +1,64 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<plist version="1.0">
+<dict>
+    <key>scope</key>
+    <string>source.cs</string>
+    <key>settings</key>
+    <dict>
+        <key>indentationFoldingEnabled</key>
+        <false/>
+        <key>foldScopes</key>
+        <array>
+            <dict>
+                <key>begin</key>
+                <string>punctuation.section.block.begin</string>
+                <key>end</key>
+                <string>punctuation.section.block.end</string>
+            </dict>
+            <dict>
+                <key>begin</key>
+                <string>punctuation.section.parameters.begin</string>
+                <key>end</key>
+                <string>punctuation.section.parameters.end</string>
+            </dict>
+            <dict>
+                <key>begin</key>
+                <string>punctuation.section.brackets.begin</string>
+                <key>end</key>
+                <string>punctuation.section.brackets.end</string>
+            </dict>
+            <dict>
+                <key>begin</key>
+                <string>punctuation.section.braces.begin</string>
+                <key>end</key>
+                <string>punctuation.section.braces.end</string>
+            </dict>
+            <dict>
+                <key>begin</key>
+                <string>punctuation.section.sequence.begin</string>
+                <key>end</key>
+                <string>punctuation.section.sequence.end</string>
+            </dict>
+            <dict>
+                <key>begin</key>
+                <string>punctuation.section.group.begin</string>
+                <!-- record parameters start as a group -->
+                <key>end</key>
+                <string>punctuation.section.group.end | punctuation.section.parameters.end</string>
```

Comment:
The mentioned issue was fixed by #3320.

That said, I have a feeling those generous selectors could cause unexpected behavior in other situations. So we should rather try to fix such things in syntax definitions.

```suggestion
                <key>end</key>
                <string>punctuation.section.group.end</string>
```